modules/op-assembling-a-pipeline.adoc

```diff
@@ -0,0 +1,102 @@
+// Ths module is included in the following assembly:
+//
+//  assembly_creating-applications-with-cicd-pipelines.adoc
+
+[id="assembling-a-pipeline_{context}"]
+= Assembling a Pipeline
+
+A Pipeline represents a CI/CD flow and is defined by the Tasks to be executed. It specifies how the Tasks interact with each other and their order of execution using the `inputs` , `outputs`, and the `runAfter` parameters. It is designed to be generic and reusable in multiple applications and environments.
+
+In this section you will create a Pipeline that takes the source code of the application from GitHub and then builds and deploys it on {product-title}.
+
+//The image below displays the various components of `pipelines-tutorial` Pipeline, and how these components interact with each other.
+
+//image::op-assemble-a-pipeline.png[]
+
+
+The Pipeline performs the following tasks for the back-end application `vote-api` and front-end application `vote-ui`:
+
+* Clones the source code of the application from the Git repositories `api-repo` and `ui-repo`.
+* Builds the container images `api-image` and `ui-image` using the `buildah` clustertask.
+* Pushes the `api-image` and `ui-image` images to the internal image registry.
+* Deploys the new images on OpenShift using the `apply-manifests` and `update-deployment` Tasks.
+
+[discrete]
+.Procedure
+
+. Copy the contents of the following sample Pipeline YAML file and save it:
++
```

Comment:
s/+/[source,yaml]
Should be using source=YAML/ style for all of these file samples